tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Test case for syntax error does not verify the content of the warning message.

While the test ensures a warning is raised when encountering a syntax error, it does not verify the content of the warning message. It's important to check that the warning message accurately reflects the error encountered, including the file path and the specific syntax error message.